core/src/sql/statements/output.rs

```diff
@@ -39,7 +40,23 @@ impl OutputStatement {
 		// Fetch any
 		if let Some(fetchs) = &self.fetch {
 			for fetch in fetchs.iter() {
-				val.fetch(stk, ctx, opt, fetch).await?;
+				let i: &Idiom;
+				let new_idiom: Idiom;
+				if let Value::Idiom(idiom) = &fetch.0 {
+					i = idiom;
+				} else if let Value::Param(param) = &fetch.0 {
+					let p = param.compute(stk, ctx, opt, None).await?;
+					if let Value::Strand(s) = p {
+						let p: Part = Part::Field(Ident(s.0));
+						new_idiom = Idiom(vec![p]);
+						i = &new_idiom;
+					} else {
+						break;
+					}
+				} else {
+					break;
+				}
```

Comment:
Breaking here is probably not the right behavior, this will cause an invalid production to just ignore the fetch without giving any indication back to the user.

If the param value is not a string you should probably return an error that it must be a string, and if the value is neither an param, nor an idiom, then it the query is invalid so you should probably return the Error::Unreachable.